plugins/cmd/chainlink-solana/main.go

```diff
@@ -70,9 +79,18 @@ func (c *chainPlugin) NewRelayer(ctx context.Context, config string, keystore lo
 		return nil, fmt.Errorf("failed to create chain: %w", err)
 	}
 	r := pkgsol.NewRelayer(c.lggr, chainSet)
+
+	err = c.promServer.Start()
+	if err != nil {
+		err = fmt.Errorf("Failed to start prometheus server: %w", err)
+		c.lggr.Critical(err.Error())
+		return nil, err
+	}
```

Comment:
My last comment was wrong, but this still seems like it fits in `func main()` regardless, no? Or does that not work for some reason?

Reply:
i went back and forth on this point. i started with it in main. i remember why i moved it; perhaps i thought i would be easier to test, but that part of the story has changed since i wrote this. i can move it back

Reply:
done